### PR TITLE
Update Delete Operation to handle 'NoContent' returns

### PR DESCRIPTION
**Pull request by huangpf**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@huangpf__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (phuang). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: src/ResourceManagement/Compute/Compute.Tests/ScenarioTests/OSProfileTests.cs [hunk 1]
```diff
@@ -294,5 +320,71 @@ public void TestVMWithLinuxOSProfile()
...
+        //    using appropriate SubscriptionName  (you can change Region, ResourceGroupName, and KeyVaultName too)
+        // 2. Update static values below (e.g. VaultId and CertifcateUrl) with values returned when running the script
+        // 3. Complete the recording. 
+        // 4. After recording completes, please delete key vault you created by deleting the whole resource group.
+
+        public static string KeyVaultId = "/subscriptions/0e1dad42-1d4c-4329-9d38-ea44c03a519d/resourceGroups/pslibtestosprofile/providers/Microsoft.KeyVault/vaults/pslibtestkeyvault";
+
+        public static Uri CertificateUrl = new Uri("https://pslibtestkeyvault.vault.azure.net:443/secrets/WinRM/5c0b61f14941409aa752665210fd1319");
+
+#pragma warning disable 1998
```
Comment (huangpf) [inline src/ResourceManagement/Compute/Compute.Tests/ScenarioTests/OSProfileTests.cs]: @yugangw-msft It's actually pretty common to disable warning 1998 in the build environment. Is it feasible for you to disable it at the SDK solution level? Thanks.

File: src/ResourceManagement/Compute/Compute.Tests/Resources/WinRMCertificate.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
...PaiqqSUCC9b0Fm1ABqu1UZOIBE+EK/21S8LAj5EfEsE4iK0jc1z1PoLk2MhW2N0bBGtJPaJ3roaqZtMraa6cNn/hb3LN7ynIh0QVM8kl13Dkqqb6mJsBd2fJ5QvAeAa0ytq9U2rAwlyjPz5DbqY5FcZQ7xMeb42Khw8jl/6K2HVNo8MrSP+s6qz9ekcdoqQr0x5BIS0yF3YljfUde+u4cRUkksVExHeT+01Dz3mQV95mQrds4NvdCtDlQfg/u7bzXG1NEKVk742PSnDhM3kKO+5e6D58k0Q2BpJJm/Iea+qCMc6c0Ff2DnvTT0ExlZ4JErynosEjhlS1fzjfdYxAv+uq6GE0561Tg4ajlfKc+M8Up0GYA+CT3gYdq7iwvzuflHKyAauk1XmplYJB36xj1RYHgFZfOoGog7iAfGeFggdnfpoidsRkm9PB9JLYok8+jz3KjNsJaYIm+3oWyvqkdaVX+6oVLGODwkX6S2N+Kd5iPCxBWD7THI8nUHqGM7QkaxnwlR5r9Dp3Mm+/L6Ujopg6fcbJsZK2RwBkbykB8AXbi2nRC7OTmmOWqeXVi2U/Ldjtbtfp9jrQ9qDnjyR/L7qmBocpk/A31o8qR3yaFGgdqPfmMT8LFQkPKnw1s92Kbx3GOvCHIbLO4KU5xLC6anSMAq8Z8MEvz2zonYwDR3PvAQch9RjFxXa7gETG6Iq73HOx2zN6dZnRn5VNfFe6UAG2MTZmIMuvmXx21bmn8fZj3d59CLpZSAK0ZwCCSMF1VbsMk51wV6yZE7J7y7xWVJyP/Xo5E4oJU7qKst7OdxFzXwU9Z/qWpLrgXPTbaeWrFZa8o7b2PY5UbksEcET0YFB58+CDYpf/cxtXJ2VoEXyVKseEkjf9RCwucVRyJcadSCrKhfvd2ZVPt9rt932Llvn/k/L5FCbxaCzhFTfGXwQQvkiJcVWc/gWNP8PrzIyPMR1JGro9ZvnwLK/eO+ObTeSuiryYaKZunXhQkuSUh8sJWulW8rNJ6EBnh+sJdWZviAmZ9/gLdGt9KP0Jw/Lca39O0C2N+lxd/RNHflFYLN5Yof9nptg8cq/z5rbkMrk4V8Sa3gZYEol2LEvRhhsngLbosb7UU4WjhKMDcwHzAHBgUrDgMCGgQU7GcfdYKQVdbY6NmukAv3JkdplwYEFGvOlSHr5lprrehexivC8lKW9rpU",
```
Comment (yugangw-msft) [inline src/ResourceManagement/Compute/Compute.Tests/Resources/WinRMCertificate.json]: so this credential is fine to expose to the public, right?

Comment (huangpf) [inline src/ResourceManagement/Compute/Compute.Tests/Resources/WinRMCertificate.json]: Yes, already confirmed.

File: src/ResourceManagement/Compute/Compute.Tests/ScenarioTests/OSProfileTests.cs (hunk 1 above)
Comment (yugangw-msft) [inline src/ResourceManagement/Compute/Compute.Tests/ScenarioTests/OSProfileTests.cs]: sure, i can do

Comment (huangpf) [inline src/ResourceManagement/Compute/Compute.Tests/ScenarioTests/OSProfileTests.cs]: Thanks!

**Comment by yugangw-msft** [comment]
do you need to pump up the nuget version?

**Comment by yugangw-msft** [comment]
lgtm

**Pull request merged**